pkg/webhook/server.go

```diff
@@ -54,14 +56,16 @@ func NewServer(name, namespace string, port int, cm *certificate.Manager, handle
 		},
 	}
 	// set the certificate once, which will also call onCertChange.
-	if err := s.setCert(name, namespace); err != nil {
+	if err := s.setCert(); err != nil {
```

Comment:
Should we subscribe to the cert rotation pub sub here before ever issuing the webhook cert so we don't miss any rotations?

Reply:
good catch, I added a test case that demonstrates this issue and working through a solution.

Reply:
thoughts?

Reply:
Could we add a health probe for the validating webhook server to indicate if we failed to issue the certificate and therefore failed to run the server?

Reply:
good idea, adding health probe would be helpful in both scenarios

Reply:
I'm not sure if there is more context around why we haven't chosen to do this in the past. @steeling do you know?

Reply:
there's a handful of places we silently ignore failures. This one is seems more critical in that we should panic on an error

Reply:
I tested this out some more and found I was wrong. the `events.GenericEventRecorder().FatalEvent` forces a webhook shut down so it goes into a crashloop.  I've updated everything to continue to work that way.